### DISCUSSION TITLE
Enable Active Learning in the Training phase

### DISCUSSION THREAD

**Discussion by mehrdadasadiut** [open-edge-platform/anomalib#2372] [Feature Requests]
### What is the motivation for this task?

I would like to have some features that could enable a user to be in the loop within anomaly detection.

### Describe the solution you'd like

I would like to have some features that could enable a user to be in the loop within anomaly detection.

### Additional context

maybe in the context of PatchCore algorithm, how easy is to have different strategies of active learning applied?

**Comment by samet-akcay** [comment]
Hi @mehrdadasadiut, do you have any papers or active learning approaches in mind?

**Reply by mehrdadasadiut** [reply]
Hi Samet, thanks for your response. Since PatchCore works with one normal class, I am not sure how to incorporate active learning since the oracle/human user may also incorporate (label) abnormal examples.
I think that it is good to have some pool-based active learning that actively involve a user / an oracle to label examples that the anomaly detection model is not confident about.
or decision boundary adaptation such as the paper "Active anomaly detection based on deep one-class classification".
Do you think it is feasible to have such feature enabled?
thanks for your contribution.

**Reply by mehrdadasadiut** [reply]
Dear Samet, is it doable?

**Reply by samet-akcay** [reply]
it is doable but we currently do not have it on our short-term roadmap. 
The closest solution that we could have in near future is SuperSimpleNet paper that @blaz-r will implement in anomalib soon
https://github.com/blaz-r/SuperSimpleNet

**Reply by blaz-r** [reply]
While the SuperSimpleNet can support active learning (AL), it has more to do with the **framework** than with the model itself. 

AL can be applied to most supervised methods (even unsupervised - but not with anomalous samples) but that would require setting up the AL framework including a special datamodule, query strategies and a custom training loop. 
Datamodule needs to be modified in a way to support handling of unlabelled and labelled samples. We'd also need to implement query strategies used to select new samples to be labelled. The entire thing then also needs a custom training-querying-label-repeat loop.

As Samet said, this is currently not on the short-term roadmap, but if any contributor really wants to work on that I'm sure it can be arranged 😉
